### PR TITLE
Add documentation for enabling gRPC logging

### LINKED ISSUE #1907: Document Logging of Google Cloud libraries.
Logging in most Google Cloud libraries is possible by enabling the logging of the lower-level libraries used to make the API calls. These libraries are GRPC and Google API Client.

```ruby
require "logger"

app_logger = Logger.new $stderr
app_logger.level = Logger::WARN

# Set GRPC logger
GRPC::DefaultLogger::LOGGER = app_logger

# Set the Google API Client logger
Google::Apis.logger = app_logger
```

Logging can be very useful for debugging applications, or performance tuning. We think this should be documented somewhere that users can more easily discover which to choose for the library they are using.

See #1216 for more discussion about this.

### PR DESCRIPTION
**Pull request by quartzmo**
[closes #1907]

### CODE REVIEW

**Comment by quartzmo** [comment]
@blowmage I tried setting the logger as you suggested in #1907, but I got this warning:

```sh
irb(main):006:0> GRPC::DefaultLogger::LOGGER = my_logger
(irb):6: warning: already initialized constant GRPC::DefaultLogger::LOGGER
```

So instead I adapted the suggested example in the gRPC [spec_helper.rb](https://github.com/grpc/grpc/blob/master/src/ruby/spec/spec_helper.rb). It works, but can anyone suggest a simplification?

**Comment by blowmage** [comment]
I agree it should not be as difficult as it is to set a custom Logger. Ideally it should be simply:

```ruby
GRPC.logger = my_logger
```

File: google-cloud-datastore/README.md [hunk 1]
```diff
@@ -47,6 +47,31 @@ query = datastore.query("Task").
...
+Configuring a Ruby stdlib logger:
+
+```ruby
+require "logger"
+
+my_logger = Logger.new $stderr
+my_logger.level = Logger::WARN
+
+module MyLogger
+  LOGGER = my_logger
```
[the remark this answers is not in the document]
Comment (quartzmo) [reply google-cloud-datastore/README.md]: Unless it's bound to the default scope in IRB, where I tested it?

**Comment by quartzmo** [comment]
@dazuma Updated, PTAL.

**Comment by quartzmo** [comment]
Thanks for approving, I will copy this change to the READMEs (and "guide" docs where appropriate) for all gRPC packages, then ping for a second look.

**Comment by quartzmo** [comment]
@jbolinger I opened googleapis/gapic-generator#2022, "Ruby: Add logging configuration instructions to README and guide docs".

**Comment by quartzmo** [comment]
@jbolinger PTAL, I just added two commits, for changes we discussed in googleapis/gapic-generator#2022 :

- Move logging docs below Next Steps in READMEs and guides.
- Add Supported Ruby Versions above Next Steps in READMEs if missing.

**Comment by jbolinger** [comment]
@quartzmo why did you move the version section above the next steps? I thought we said below and at `##`, like this: https://gist.github.com/jbolinger/0bcc9b62d4ee84a59fd454c8cc347f6f

**Comment by quartzmo** [comment]
Ah, sorry, I didn't understand. I did it because in a number of the generated READMEs, such as in [dataproc for example](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-dataproc/README.md#supported-ruby-versions) it is above Next Steps. So, I thought it must be need to be there. However, I'm happy to move it down as in your gist. Will update.

**Comment by quartzmo** [comment]
@jbolinger Updated, PTAL. I will squash commits before merging.

**Comment by jbolinger** [comment]
@quartzmo looks good!

**Pull request merged**